### PR TITLE
Allow for definition of custom ingress / egress gateways

### PR DESCRIPTION
**Pull request by jwbernin**
This MR allows the user to create custom ingress and egress gateway definitions in the ingress.yaml and egress.yaml files, then enable them on the SMCP object being deployed by toggling the appropriate values in the values.yaml file.

### CODE REVIEW

File: charts/install_smcp_with_helm/control-plane/templates/servicemeshcontrolplane-basic-install.yaml [hunk 1]
```diff
@@ -58,11 +58,19 @@ spec:
               targetPort: 15443
           type: ClusterIP
     egress:
+{{- if .Vaules.smcp_enable_custom_egress_config }}
```
Comment (willowmck) [inline charts/install_smcp_with_helm/control-plane/templates/servicemeshcontrolplane-basic-install.yaml]: typo in Values

File: charts/install_smcp_with_helm/control-plane/templates/servicemeshcontrolplane-basic-install.yaml [hunk 2]
```diff
@@ -58,11 +58,19 @@ spec:
...
+{{ .Files.Get smcp_custom_egress_config_file | indent 6 }}
+{{- else }}
       enabled: false
       service: {}
+{{- end }}
     ingress:
+{{- if .Vaules.smcp_enable_custom_ingress_config }}
```
Comment (willowmck) [inline charts/install_smcp_with_helm/control-plane/templates/servicemeshcontrolplane-basic-install.yaml]: typo in Values

File: charts/install_smcp_with_helm/control-plane/templates/servicemeshcontrolplane-basic-install.yaml (hunk 1 above)
Comment (jwbernin) [reply charts/install_smcp_with_helm/control-plane/templates/servicemeshcontrolplane-basic-install.yaml]: well, at least I know copy/paste works...

fixed in second commit.

**Comment by willowmck** [comment]
Looks like there are conflicts now.  You'll need to merge from main to resolve them.

**Pull request merged**